cdap-cli-tests/src/test/java/co/cask/cdap/cli/CLIMainTest.java

```diff
@@ -436,6 +441,71 @@ public void testRuntimeArgs() throws Exception {
+      // Call non-version service and get response from service with Default version
+      testCommandOutputContains(cli, "call service " + nonVersionServiceId + " POST /echo body \"testBody\"",
+                                String.format("%s:testBody", ApplicationId.DEFAULT_VERSION));
+
+      // Start serviceV1
+      Map<String, String> runtimeArgs1 = ImmutableMap.of("sdf", V1_SNAPSHOT);
+      String runtimeArgs1KV = spaceJoiner.withKeyValueSeparator("=").join(runtimeArgs1);
+      testCommandOutputContains(cli, "start service " + qualifiedServiceIdV1 + " '" + runtimeArgs1KV + "'",
+                                "Successfully started service");
+      assertProgramStatus(programClient, serviceV1, "RUNNING");
+
+      // Set RouteConfig to route all traffic to service with default version
+      Map<String, Integer> routeConfig = ImmutableMap.of(ApplicationId.DEFAULT_VERSION, 100, V1_SNAPSHOT, 0);
+      String routeConfigKV = spaceJoiner.withKeyValueSeparator("=").join(routeConfig);
+      testCommandOutputContains(cli, "set routeconfig service " + nonVersionServiceId + " '" + routeConfigKV + "'",
+                                "Successfully set route configuration");
+      for (int i = 0; i < 10; i++) {
+        testCommandOutputContains(cli, "call service " + nonVersionServiceId + " POST /echo body \"testBody\"",
+                                  String.format("%s:testBody", ApplicationId.DEFAULT_VERSION));
+      }
+      // Get the RouteConfig set previously
+      testCommandOutputContains(cli, "get routeconfig service " + nonVersionServiceId, GSON.toJson(routeConfig));
+
+      // Set RouteConfig to route all traffic to serviceV1
+      routeConfig = ImmutableMap.of(ApplicationId.DEFAULT_VERSION, 0, V1_SNAPSHOT, 100);
+      routeConfigKV = spaceJoiner.withKeyValueSeparator("=").join(routeConfig);
+      testCommandOutputContains(cli, "set routeconfig service " + nonVersionServiceId + " '" + routeConfigKV + "'",
+                                "Successfully set route configuration");
+      for (int i = 0; i < 10; i++) {
+        testCommandOutputContains(cli, "call service " + nonVersionServiceId + " POST /echo body \"testBody\"",
+                                  String.format("%s:testBody", V1_SNAPSHOT));
+      }
+      // Get the RouteConfig set previously
+      testCommandOutputContains(cli, "get routeconfig service " + nonVersionServiceId, GSON.toJson(routeConfig));
+      // Delete the RouteConfig and verify the RouteConfig is empty
+      testCommandOutputContains(cli, "delete routeconfig service " + nonVersionServiceId,
+                                "Successfully delete route configuration");
+      testCommandOutputContains(cli, "get routeconfig service " + nonVersionServiceId, GSON.toJson(ImmutableMap.of()));
+    } finally {
+      testCommandOutputContains(cli, "stop service " + qualifiedServiceId, "Successfully stopped service");
```

Comment:
`programClient.stopAll(NamespaceId.DEFAULT.toId());`
can be used to stop only those that are running.